### ISSUE TITLE
[@typescript-eslint/no-unused-vars] incorrect report on dotted-namespace

### ISSUE THREAD

**Issue by ueokande** [typescript-eslint/typescript-eslint#2589]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": "error"
  }
}
```

```TS
declare namespace foo.bar {
  type str = string;
}

console.log("Hello!" as foo.bar.str);
```

**Expected Result**

No error because both namespace and type are used in the code.

ESLint reports no-unused-vars when I use `declare namespace` with dotted-namespace. If non-dotted namespace such as `foo` and used as `"Hello!" as foo.str`,  eslint reports nothing.  I can see this issue since v4.0.0, and cannot reproduce on v3.10.0.

**Actual Result**

>  1:23  error  'bar' is defined but never used  @typescript-eslint/no-unused-vars
>  2:8   error  'str' is defined but never used  @typescript-eslint/no-unused-vars

**Additional Info**

None

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.2.0` |
| `@typescript-eslint/parser`        | `4.2.0` |
| `TypeScript`                       | `4.0.3` |
| `ESLint`                           | `7.9.0` |
| `node`                             | `12.13.1` |

**Comment by HolgerJeromin** [comment]
duplicate of #2573

**Comment by ueokande** [comment]
thanks
